provisioning_templates/snippet/preseed_networking_netplan_setup.erb

```diff
@@ -0,0 +1,62 @@
+<%#
+kind: snippet
+name: preseed_networking_netplan_setup
+model: ProvisioningTemplate
+snippet: true
+description: this will configure host networking on Ubuntu 18.04+ using Netplan.io, it configures your primary interface as well
+    as other configures NICs. It supports physical, VLAN and Alias interfaces. It's intended to be
+    called in your preseed finish template.
+%>
+<% host_subnet = @host.subnet -%>
+<% host_dhcp = host_subnet.nil? ? false : host_subnet.dhcp_boot_mode? -%>
+<% host_subnet6 = @host.subnet6 -%>
+<% host_dhcp6 = host_subnet6.nil? ? false : host_subnet6.dhcp_boot_mode? -%>
+
+real=`ip -o link | awk '/<%= @host.mac -%>/ {print $2;}' | sed s/://`
+cat << EOF > /etc/netplan/01-netcfg.yaml
+# This file describes the network interfaces available on your system
+# For more information, see netplan(5).
+network:
+  version: 2
+  ethernets:
+    #<%= @host.primary_interface.identifier %>
+    $real:
+<% if host_dhcp6 -%>
+      dhcp6: true
+<% end -%>
+<% if host_dhcp -%>
+      dhcp4: true
+<% else -%>
+      addresses: [ <%= @host.ip %>/<%= host_subnet.cidr %><% if @host.ip6 && host_subnet6 && !host_dhcp6 -%>, <%= @host.ip6 %>/<%= host_subnet6.cidr %><% end -%> ]
+      gateway4: <%= host_subnet.gateway %>
+      <% if host_subnet6.gateway -%>gateway6: <%= host_subnet6.gateway %><% end %>
```

Comment:
You probably need to check if `host_subnet6` exists before using it. That's the likely source

Reply:
Great catch. Thanks.